guides/mastering_transceivers.md

```diff
@@ -0,0 +1,536 @@
+:ok = PeerConnection.set_remote_description(pc2, offer)
+{:ok, answer} = PeerConnection.create_answer(pc2)
+:ok = PeerConnection.set_local_description(pc2, answer)
+:ok = PeerConnection.set_remote_description(pc1, answer)
+
+{:ok, offer} = PeerConnection.create_offer(pc1)
+:ok = PeerConnection.set_local_description(pc1, offer)
+:ok = PeerConnection.set_remote_description(pc2, offer)
+
+dbg(PeerConnection.get_transceivers(pc2))
+
+{:ok, answer} = PeerConnection.create_answer(pc2)
+:ok = PeerConnection.set_local_description(pc2, answer)
+
+dbg(PeerConnection.get_transceivers(pc2))
+
+:ok = PeerConnection.set_remote_description(pc1, answer)
+```
+
+<!-- tabs-close -->
+
+## Recycling m-lines
+
+When calling stop on an `RTCRtpTransceiver`, it will eventually be removed from
+the connection's set of transceivers.
+However, the number of m-lines in SDP offer/answer can never decrease.
+`m-lines` corresponding to stopped transceivers can be reused when a new transceiver appears.
+This process is known as recycling m-lines, and it prevents SDP from becoming excessively large.
+
+Things to note:
+* A new transceiver will always attempt to reuse the first free m-line, regardless of its kind i.e.,
+whether it's audio or video
+* The order of transceivers in a connection's set of transceivers matches the order in which
+the transceivers were added, but it may be different than the order of m-lines in SDP offer/answer.
+
+<!-- tabs-open -->
+
+### JavaScript
+
+[![JS FIDDLE](https://img.shields.io/badge/-JS%20FIDDLE-blueviolet)](https://jsfiddle.net/mickel8/m7yq3fv0/)
+
+```js
+pc1 = new RTCPeerConnection();
+pc2 = new RTCPeerConnection();
+
+tr1 = pc1.addTransceiver("audio");
+tr2 = pc1.addTransceiver("video");
+
+offer = await pc1.createOffer();
+await pc1.setLocalDescription(offer);
+await pc2.setRemoteDescription(offer);
+answer = await pc2.createAnswer();
+await pc2.setLocalDescription(answer);
+await pc1.setRemoteDescription(answer);
+
+tr1.stop();
+
+offer = await pc1.createOffer();
+await pc1.setLocalDescription(offer);
+await pc2.setRemoteDescription(offer);
+answer = await pc2.createAnswer();
+await pc2.setLocalDescription(answer);
+await pc1.setRemoteDescription(answer);
+
+// notice that after renegotiation
+// pc1.getTransceivers() will only
+// return one (video) transceiver
+
+tr3 = pc1.addTransceiver("video");
+
+// Notice that createOffer will reuse (recycle)
+// free m-line, even though its initiall type was audio.
+// However, pc1.getTransceivers() will return [tr1, tr3].
+// That's important as the order of transceivers doesn't
+// have to match the order of m-lines i.e. tr3 maps to m-line
+// with index 0 and tr1 maps to m-line with index 1.
+offer = await pc1.createOffer();
+await pc1.setLocalDescription(offer);
+await pc2.setRemoteDescription(offer);
+answer = await pc2.createAnswer();
+await pc2.setLocalDescription(answer);
+await pc1.setRemoteDescription(answer);
+```
+
+### Elixir WebRTC
```

Comment:
I was wondering, maybe we should just use `Elixir` for the code blocks? It's not like there's an alternative 🙂

Reply:
Yeah I know, but Elixir WebRTC looks cool imo